ENIGMAsystem/SHELL/Universal_System/roomsystem.cpp

```diff
@@ -250,23 +250,18 @@ namespace enigma
         dit->second.tiles.clear();
       }
     }
-    for (int i = 0; i < tilecount; i++) {
+    for (size_t i = 0; i < tiles.size(); i++) {
       tile t = tiles[i];
       drawing_depths[t.depth].tiles.push_back(t);
     }
     load_tiles();
     //Tiles end
 
-    object_basic* is[instancecount];
-    for (int i = 0; i < instancecount; i++) {
-      inst *obj = &instances[i];
-      object_basic *existing;
-      if ((existing = enigma::fetch_instance_by_id(obj->id))) {
-        is[i] = NULL;
-      } else {
-        is[i] = instance_create_id(obj->x,obj->y,obj->obj,obj->id);
-      }
-    }
+    std::vector<object_basic*> created;
+    created.reserve(instances.size());
+    for (const inst &obj : instances)
+      if (!enigma::fetch_instance_by_id(obj.id))
+        created.emplace_back(instance_create_id(obj.x,obj.y,obj.obj,obj.id));
```

Comment:
JFYI `emplace_back` and `push_back` for an int vector do the same thing.

Reply:
That's what I thought, but oh well, better to play it safe than sorry in case someone changes it.